reconcile/test/test_secret_reader.py

```diff
@@ -1,11 +1,117 @@
+from unittest.mock import MagicMock
 import pytest
 
 import reconcile.utils.secret_reader
 from reconcile.utils import vault
-from reconcile.utils.secret_reader import SecretReader, SecretNotFound
+from reconcile.gql_definitions.fragments.vault_secret import VaultSecret
+from reconcile.utils.secret_reader import (
+    ConfigSecretReader,
+    SecretReader,
+    SecretNotFound,
+    VaultSecretReader,
+)
 from reconcile.utils.vault import _VaultClient
 
 
+VAULT_READ_EXPECTED = {"key": "value"}
+VAULT_READ_ALL_EXPECTED = {"key2": "value2"}
+
+
+@pytest.fixture
+def vault_mock():
+    read_mock = MagicMock()
+    read_mock.side_effect = [VAULT_READ_EXPECTED] * 100
+    read_all_mock = MagicMock()
+    read_all_mock.side_effect = [VAULT_READ_ALL_EXPECTED] * 100
+    vault_mock = MagicMock(spec=_VaultClient)
```

Comment:
Any reason not to use `create_autospec`? tldr: autospec is recursive and I don't think `spec` is? There's more in the `mock` documentation.

Reply:
Good point! Thank you :bow: